src/lib.rs

```diff
@@ -379,22 +409,31 @@ impl Build {
             build.define("ZMQ_HAVE_UIO", "1");
 
             if target.contains("android") {
-                build.define("ZMQ_HAVE_STRLCPY", "1");
+                has_strlcpy = true;
             }
 
             if target.contains("musl") {
-                build.define("ZMQ_HAVE_STRLCPY", "1");
+                has_strlcpy = true;
             }
         } else if target.contains("apple") {
             create_platform_hpp_shim(&mut build);
             build.define("ZMQ_IOTHREAD_POLLER_USE_KQUEUE", "1");
             build.define("ZMQ_POLL_BASED_ON_POLL", "1");
             build.define("HAVE_STRNLEN", "1");
-            build.define("ZMQ_HAVE_STRLCPY", "1");
             build.define("ZMQ_HAVE_UIO", "1");
             build.define("ZMQ_HAVE_IPC", "1");
+            has_strlcpy = true;
         }
 
+        // https://github.com/jean-airoldie/zeromq-src-rs/issues/28
+        #[cfg(target_env = "gnu")]
+        if !has_strlcpy && glibc::has_strlcpy() {
```

Comment:
I think this `cfg` checks the `target_env` of the host. Adding this snippet right above:

```rust
        panic!(
            "CARGO_CFG_TARGET_ENV: {:?} target_env=gnu?: {}",
            env::var("CARGO_CFG_TARGET_ENV"),
            cfg!(target_env = "gnu")
        );
```

With output:

```
  thread 'main' panicked at 'CARGO_CFG_TARGET_ENV: Ok("msvc") target_env=gnu?: true', src/lib.rs:429:9
```
confirms that.